ste/sourceInfoProvider-Local_linux.go

```diff
@@ -0,0 +1,189 @@
+// +build linux
+
+package ste
+
+import (
+	"errors"
+	"github.com/Azure/azure-storage-azcopy/v10/common"
+	"golang.org/x/sys/unix"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func (f localFileSourceInfoProvider) HasUNIXProperties() bool {
+	return true
+}
+
+func (f localFileSourceInfoProvider) GetUNIXProperties() (common.UnixStatAdapter, error) {
+	// Can we use statx?
+	var uname unix.Utsname
+	err := unix.Uname(&uname)
+	if err != nil {
+		return nil, err
+	}
+
+	linuxVersion := regexp.MustCompile("(\\d+\\.\\d+\\.\\d+)").FindString(string(uname.Release[:]))
+	splits := strings.Split(linuxVersion, ".")
+
+	if len(splits) != 3 {
+		return nil, errors.New("Failed to parse linux version " + linuxVersion)
+	}
+
+	versions := make([]int64, 3)
+	for k, v := range splits {
+		versions[k], _ = strconv.ParseInt(v, 10, 64)
+	}
+
+	var resp common.UnixStatAdapter
+
+	if versions[0] > 4 || (versions[0] == 4 && versions[1] > 11) { // Can we statx?
```

Comment:
this isn't a safe way to find statx support especially because many distros backport patches and don't update the version (rather they update the subversion).
Safer/easier way is to make the unix.Statx() call and if it returns ENOSYS error fallback to unix.Stat().